python-lib/dku_stress_test_center/stress_test_center.py

```diff
@@ -125,19 +146,27 @@ def compute_test_metrics(self, test: StressTest):
         clean_df_with_pred = self._clean_df.loc[test.df_with_pred.index, :]
 
         clean_y_true, clean_y_pred, clean_probas = self._get_col_for_metrics(clean_df_with_pred)
-        perturbed_y_true, perturbed_y_pred, perturbed_probas = self._get_col_for_metrics(test.df_with_pred)
-
         perf_before = self._metric.compute(clean_y_true, clean_y_pred, clean_probas)
-        perf_after = self._metric.compute(perturbed_y_true, perturbed_y_pred, perturbed_probas)
+
         common_metrics = {
-            "perf_var": (perf_before - perf_after) * (-1 if self._metric.is_greater_better() else 1),
-            "perf_before": perf_before,
-            "perf_after": perf_after
+            "perf_before": perf_before
         }
+        if test.relevant:
+            perturbed_y_true, perturbed_y_pred, perturbed_probas = self._get_col_for_metrics(test.df_with_pred)
+            perf_after = self._metric.compute(perturbed_y_true, perturbed_y_pred, perturbed_probas)
+        else:
+            perf_after = perf_before
+            common_metrics["not_relevant_explanation"] = test.not_relevant_explanation
+        common_metrics.update({
+            "perf_after": perf_after,
+            "perf_var": (perf_before - perf_after) * (-1 if self._metric.is_greater_better() else 1)
+        })
 
         extra_metrics = {}
         if test.TEST_TYPE == DkuStressTestCenterConstants.FEATURE_PERTURBATION:
-            if self.model_accessor.get_prediction_type() == DkuStressTestCenterConstants.REGRESSION:
+            if not test.relevant:
+                corruption_resilience = 1
```

Comment:
it would be great if you can add a small comment so that we won't forget why it is equal 1 in the future ...  I struggled like for 3s before understand it

Reply:
Done in  f111f51; is it clear enough?

Reply:
yes that's great!